<a href="https://colab.research.google.com/github/AshikSathiya/Project2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2

---



In [1]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#First choice: Body Fat Prediction Dataset


In [3]:
# load the dataset
fpath = '/content/drive/MyDrive/CodingDojo/Project 2/cirrhosis.csv'
df = pd.read_csv(fpath, header=None)

df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 419 entries, 0 to 418
Data columns (total 20 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       419 non-null    object
 1   1       419 non-null    object
 2   2       419 non-null    object
 3   3       313 non-null    object
 4   4       419 non-null    object
 5   5       419 non-null    object
 6   6       313 non-null    object
 7   7       313 non-null    object
 8   8       313 non-null    object
 9   9       419 non-null    object
 10  10      419 non-null    object
 11  11      285 non-null    object
 12  12      419 non-null    object
 13  13      311 non-null    object
 14  14      313 non-null    object
 15  15      313 non-null    object
 16  16      283 non-null    object
 17  17      408 non-null    object
 18  18      417 non-null    object
 19  19      413 non-null    object
dtypes: object(20)
memory usage: 65.6+ KB


0       1       2                3      4    5        6             7   \
0  ID  N_Days  Status             Drug    Age  Sex  Ascites  Hepatomegaly   
1   1     400       D  D-penicillamine  21464    F        Y             Y   
2   2    4500       C  D-penicillamine  20617    F        N             Y   
3   3    1012       D  D-penicillamine  25594    M        N             N   
4   4    1925       D  D-penicillamine  19994    F        N             Y   

        8      9          10           11       12      13        14      15  \
0  Spiders  Edema  Bilirubin  Cholesterol  Albumin  Copper  Alk_Phos    SGOT   
1        Y      Y       14.5          261      2.6     156      1718  137.95   
2        Y      N        1.1          302     4.14      54    7394.8  113.52   
3        N      S        1.4          176     3.48     210       516    96.1   
4        Y      S        1.8          244     2.54      64    6121.8   60.63   

              16         17           18     19  
0  Tryglicerides  Platelets  Prothrombin  Stage  
1            172        190         12.2      4  
2             88        221         10.6      3  
3             55        151           12      4  
4             92        183         10.3      4

##1. Source of data


This dataset comes from Kaggle

##2. Brief description of data



Cirrhosis is an advanced liver condition caused by various liver diseases, including hepatitis and chronic alcoholism. A Mayo Clinic trial focused on primary biliary cirrhosis (PBC) collected data from 424 patients referred to the clinic over ten years. Among them, 312 patients took part in a randomized drug trial, while the remaining 112 patients consented to basic measurements and survival monitoring. However, six of the latter group were lost to follow-up, leaving data on 106 non-participating cases and the 312 participants in the randomized trial. For more details, refer to Fleming and Harrington's book and articles by Dickson et al. and Markus et al. published in 1989.










##3. What is the target?



Status: status of the patient C (censored), CL (censored due to liver tx), or D (death)

##4. What does one row represent? (A person? A business? An event? A product?)



One row in the dataset represents a single individual patient who participated in the drug trial

##5. Is this a classification or regression problem?



This is a classification problem

##6. How many features does the data have?



19 features

##7. How many rows are in the dataset?



419 rows

##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?



Missing Data: Addressing missing values appropriately.

Categorical Variables: Converting categorical attributes into suitable formats.

Imbalanced Classes: Handling imbalanced outcomes in "Status"

Outliers: Identifying and managing outliers in the data.

Feature Selection: Selecting relevant attributes for modeling.

Data Normalization/Scaling: Normalizing or scaling numerical data for fair comparisons.

Overfitting: Preventing overfitting by using proper validation and regularization.


#Second choice: Student Mental Health



In [4]:
# load the dataset
fpath1 = '/content/drive/MyDrive/CodingDojo/Project 2/wines_SPA.csv'
df1 = pd.read_csv(fpath1, header=None)

df1.info()
df1.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7501 entries, 0 to 7500
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       7501 non-null   object
 1   1       7501 non-null   object
 2   2       7499 non-null   object
 3   3       7501 non-null   object
 4   4       7501 non-null   object
 5   5       7501 non-null   object
 6   6       7501 non-null   object
 7   7       7501 non-null   object
 8   8       6956 non-null   object
 9   9       6332 non-null   object
 10  10      6332 non-null   object
dtypes: object(11)
memory usage: 644.7+ KB


0              1     2       3            4        5   \
0         winery           wine  year  rating  num_reviews  country   
1  Teso La Monja          Tinto  2013     4.9           58   Espana   
2         Artadi  Vina El Pison  2018     4.9           31   Espana   
3   Vega Sicilia          Unico  2009     4.8         1793   Espana   
4   Vega Sicilia          Unico  1999     4.8         1705   Espana   

                 6       7                     8     9        10  
0            region   price                  type  body  acidity  
1              Toro     995              Toro Red     5        3  
2    Vino de Espana   313.5           Tempranillo     4        2  
3  Ribera del Duero  324.95  Ribera Del Duero Red     5        3  
4  Ribera del Duero  692.96  Ribera Del Duero Red     5        3

## New Section

##1. Source of data


Kaggle

##2. Brief description of data



This dataset focuses on red variants of Spanish wines, exploring popularity and description metrics and their impact on wine quality. The dataset is suitable for classification or regression tasks. The quality classes are ordered and imbalanced, ranging from nearly 5 to 4 points. The main objective is to predict either the wine quality or prices based on the provided data.










##3. What is the target?



Price / Quality

##4. What does one row represent? (A person? A business? An event? A product?)



one row in the dataset represents a specific red variant of Spanish wine

##5. Is this a classification or regression problem?



Classification of preidicting quality/ Regression if predicting wine cost

##6. How many features does the data have?



10 features

##7. How many rows are in the dataset?



7500 rows

##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?



Data Cleaning: Addressing missing values and inconsistencies.

Feature Selection: Choosing relevant attributes to avoid overfitting.

Imbalanced Classes: Handling imbalanced wine quality categories.

Correlation and Multicollinearity: Managing attribute relationships.

Data Scaling: Scaling numerical features for fair comparisons.

Outliers: Treating outliers to avoid model bias.
